Source/Urho3D/Input/PointerAdapter.h

```diff
@@ -0,0 +1,143 @@
+//
+// Copyright (c) 2023-2023 the rbfx project.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+#pragma once
+
+#include "../Core/Object.h"
+#include "../Input/DirectionAggregator.h"
+
+namespace Urho3D
+{
+
+namespace PointerAdapterDetail
+{
+
+enum class SubscriptionMask : unsigned
```

Comment:
General thought: why do we hide enum and expose bool flags, when we can just expose enum and *not* have 10 separate set/get functions?

Reply:
It feels like it suits Urho codebase better. If you prefer to change that we need to change other adapters too.